src/DockManager.cpp

```diff
@@ -823,6 +824,23 @@ void CDockManager::loadPerspectives(QSettings& Settings)
 	Settings.endArray();
 }
 
+CDockWidget* CDockManager::centralWidget()
+{
+    return d->CentralWidget;
+}
+
+//============================================================================
+CDockAreaWidget* CDockManager::setCentralWidget(CDockWidget* widget)
+{
+    if(d->CentralWidget)
+    {
+        addDockWidget(RightDockWidgetArea, d->CentralWidget);
```

Comment:
Could you please explain, what you do here. If there is already a CentralWidget then you move it into the RightDockWidgetArea and make the new widget the central one - is this right?